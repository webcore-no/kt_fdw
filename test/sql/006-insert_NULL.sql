create server kt_server foreign data wrapper kt_fdw;
create foreign table kt_test (key text, value text, flag text) server kt_server;
create user mapping for public server kt_server;
delete from kt_test;
insert into kt_test values ('key1', NULL);
select * from kt_test;
insert into kt_test values ('key1', 'bar');
select * from kt_test;
insert into kt_test values ('key1', NULL);
select * from kt_test;
insert into kt_test values (NULL, NULL);
select * from kt_test;
insert into kt_test values (NULL, 'bar');
select * from kt_test;
