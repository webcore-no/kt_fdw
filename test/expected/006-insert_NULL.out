set client_min_messages=ERROR;
drop server if exists kt_server cascade;
set client_min_messages=NOTICE;
create server kt_server foreign data wrapper kt_fdw;
create foreign table kt_test (key text, value text, flag text) server kt_server;
create user mapping for public server kt_server;
delete from kt_test;
insert into kt_test values ('key1', NULL);
ERROR:  src/kt_fdw.c:ktExecForeignInsert():1486 Value is NULL, to overwrite value use 'kt_set' flag
select * from kt_test;
 key | value | flag 
-----+-------+------
(0 rows)

insert into kt_test values ('key1', 'bar');
select * from kt_test;
 key  | value | flag 
------+-------+------
 key1 | bar   | 
(1 row)

insert into kt_test values ('key1', NULL);
ERROR:  src/kt_fdw.c:ktExecForeignInsert():1486 Value is NULL, to overwrite value use 'kt_set' flag
select * from kt_test;
 key  | value | flag 
------+-------+------
 key1 | bar   | 
(1 row)

insert into kt_test values (NULL, NULL);
ERROR:  src/kt_fdw.c:ktExecForeignInsert():1397 can't get key value
select * from kt_test;
 key  | value | flag 
------+-------+------
 key1 | bar   | 
(1 row)

insert into kt_test values (NULL, 'bar');
ERROR:  src/kt_fdw.c:ktExecForeignInsert():1397 can't get key value
select * from kt_test;
 key  | value | flag 
------+-------+------
 key1 | bar   | 
(1 row)

